telegram/telegram.py

```diff
@@ -0,0 +1,194 @@
+import json
+
+from treq.client import HTTPClient
+
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.web import http
+from twisted.web.client import Agent
+from twisted.web._newclient import ResponseFailed
+
+from vumi.transports.httprpc.httprpc import HttpRpcTransport
+from vumi.config import ConfigText
+from vumi import log
+
+
+class TelegramTransportConfig(HttpRpcTransport.CONFIG_CLASS):
+    bot_username = ConfigText(
+        'The username of our Telegram bot',
+        static=True,
+        required=True,
+    )
+    bot_token = ConfigText(
+        "Our bot's unique token to access the Telegram API",
+        static=True,
+        required=True,
+    )
+    outbound_url = ConfigText(
+        'The URL our bot should make requests to',
+        default='https://api.telegram.org/bot', static=True,
+    )
+    inbound_url = ConfigText(
+        'The URL our transport will listen on for Telegram updates',
+        static=True, required=True,
+    )
+
+
+class TelegramTransport(HttpRpcTransport):
+    """
+    Telegram transport for Vumi
+    """
+    transport_type = 'telegram'
+    transport_name = 'telegram_transport'
+
+    CONFIG_CLASS = TelegramTransportConfig
+
+    @classmethod
+    def agent_factory(cls):
+        """For swapping out the Agent for use in tests"""
+        return Agent(reactor)
+
+    @inlineCallbacks
+    def setup_webhook(self):
+        # NOTE: Telegram currently only supports ports 80, 88, 443 and 8443 for
+        #       webhook setup
+        url = '%s/setWebhook' % self.outbound_url.rstrip('/')
```

Comment:
Is `rstrip('/')` actually needed here given that `self.outbound_url` is set to `'%s%s' % (self.config.outbound_url.rstrip('/'), self.config.bot_token)`?

I would create a helper method for constructing these URLs so that whatever the logic is, we only need to write it once (e.g. `url = self.telegram_api_url('setWebhook')`).